Fix render_test_results if condition on always()

zhouzhuojie opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #62997

Fixes #62979, changed the condition to listen on the previous'
job's result to be either 'success' or 'failure'.

Notice that 'skipped' will also skip this job, which is what
we want.

Differential Revision: [D30202598](https://our.internmc.facebook.com/intern/diff/D30202598)

zhouzhuojie committed:
Fix render_test_results if condition on always()

Fixes #62979, changed the condition to listen on the previous'
job's result to be either 'success' or 'failure'.

Notice that 'skipped' will also skip this job, which is what
we want.

[ghstack-poisoned]

zhouzhuojie commented on .github/templates/linux_ci_workflow.yml.j2:
```diff
@@ -439,8 +439,8 @@ jobs:
   # doesn't create the best experience
   render_test_results:
-    if: always()
-    needs: [generate-test-matrix, test, !{{ ciflow_config.root_job_name }}]
```
generate-test-matrix is already one of test's dependencies, so removing it.

zhouzhuojie replied:
oh, actually, let me move it back, they have needs references below

zhouzhuojie committed:
Update on "Fix render_test_results if condition on always()"


Fixes #62979, changed the condition to listen on the previous'
job's result to be either 'success' or 'failure'.

Notice that 'skipped' will also skip this job, which is what
we want.

[ghstack-poisoned]

tktrungna reviewed (approved):
not sure if it is possible to put `if` after `needs`, but it will be more readable in the sense of _the dependencies are declared and then are checked to determine if current job is skipped or not._

zhouzhuojie commented:
@zhouzhuojie has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30202598).

zhouzhuojie committed:
Update on "Fix render_test_results if condition on always()"


Fixes #62979, changed the condition to listen on the previous'
job's result to be either 'success' or 'failure'.

Notice that 'skipped' will also skip this job, which is what
we want.

Differential Revision: [D30202598](https://our.internmc.facebook.com/intern/diff/D30202598)

[ghstack-poisoned]

zhouzhuojie commented:
@zhouzhuojie has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30202598).
